src/foam/nanos/crunch/ui/CapabilityView.js

```diff
@@ -0,0 +1,105 @@
+foam.CLASS({
+  package: 'foam.nanos.crunch.ui',
+  name: 'MyView',
+  extends: 'foam.u2.View',
+  documentation: 'All purpose capability view that takes in an array of capaiblity ids and display these capabilities',
+
+  imports: [
+    'crunchController',
+    'notify'
+  ],
+
+  requires: [
+    'foam.u2.detail.SectionView'
+  ],
+
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'capabilities'
+    },
+    {
+      name: 'wizardlets',
+      documentation: 'wizardlets for the capability'
+    },
+    {
+      name: 'wizardletSectionsList',
+      documentation: `
+        sections for wizardlets
+        wizardletSectionsList[i] stores sections for wizardlets[i]
+      `
+    }
+  ],
+
+  methods: [
+    async function initE() {
+      const self = this;
+
+      this.start().addClass(this.myClass())
+        .add(this.slot(function(capabilities) {
+          return this.E().forEach(capabilities, async function(cap) {
+            // get wizardlets for the capability and all of its dependent capabilities
+            self.wizardlets = (await self.crunchController.getCapsAndWizardlets(cap)).wizCaps;
+
+            // add listeners on wizardlets
+            for ( const wizardlet of self.wizardlets ) {
+              self.addListeners(wizardlet.data);
+            }
+
+            // get all the sections associated with the wizardlets
+            self.wizardletSectionsList = self.crunchController.generateSections(self.wizardlets);
+            
+            this.add(self.slot(function(wizardletSectionsList) {
+              return this.E().forEach(wizardletSectionsList, function(sections, index) {
+                sections.map(section => (
+                  this.tag(self.SectionView, {
+                    section,
+                    data: self.wizardlets[index].data,
+                    showTitle: false
+                  })
+                ));
+              });
+            }));
+          });
+        }))
+      .end();
+    },
+
+    function addListeners(obj) {
+      if ( ! this.isFObject(obj) ) return;
```

Comment:
// This adds a listener to each wizardlet.data obj for the purpose of saving all user inputs. Releasing calling views from the responsibility.

^ want to suggest a bit of documentation to this function